x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LocalLogicalPlanOptimizer.java

```diff
@@ -162,6 +163,15 @@ else if (plan instanceof Project project) {
                     plan = new Eval(project.source(), project.child(), new ArrayList<>(nullLiteral.values()));
                     plan = new Project(project.source(), plan, newProjections);
                 }
+            } else if (plan instanceof MvExpand) {
+                // We cannot replace the target (NamedExpression) with a Literal
+                // https://github.com/elastic/elasticsearch/issues/109974
+                // Unfortunately we cannot remove the MvExpand right away, or we'll lose the output field (layout problems)
+                // TODO but this could be a follow-up optimization
+                plan = plan.transformExpressionsOnlyUp(
+                    FieldAttribute.class,
+                    f -> stats.exists(f.qualifiedName()) ? f : new Alias(f.source(), f.name(), null, Literal.of(f, null), f.id())
```

Comment:
Hm, rather than replacing the expression but keeping the MV_EXPAND, wouldn't it be cleaner to replace the MV_EXPAND by an EVAL? That might profit better from our `EVAL` optimizations, pruning rules, etc. (if they're applied after this - didn't check.) or at least be easy to understand what to expect from.

E.g.
```
FROM idx
| MV_EXPAND field

becomes (if field is missing)

FROM idx
| EVAL field = null
```

Reply:
I considered it as a follow-up. 
I didn't do it with this PR because it can be slightly tricky and I wanted to treat it as an optimization rather than a fix.
In particular, in the example you propose, `field` represents two different things:
- for MV_EXPAND it's the target; the output will have the same name but a different ID
- for EVAL it's a completely new field, that will need mask the previous one and have the same ID as the output for MV_EXPAND.